core/clustersmngr/factory.go

```diff
@@ -104,7 +135,28 @@ func NewClientFactory(fetcher ClusterFetcher, nsChecker nsaccess.Checker, logger
 		initialClustersLoad: make(chan bool),
 		scheme:              scheme,
 		newClustersPool:     clusterPoolFactory,
+		watchers:            []*ClustersWatcher{},
+	}
+}
+
+// Subscribe returns a new ClustersWatcher.
+func (cf *clientsFactory) Subscribe() *ClustersWatcher {
+	cw := &ClustersWatcher{cf: cf, Updates: make(chan ClusterListUpdate, 5)}
```

Comment:
It is not clear why this capacity was chosen for the buffer. Perhaps it can be configured?

Reply:
Yeah, this could probably be `1` rather than `5`.